packages/jest-cli/src/TestRunner.js

```diff
@@ -238,10 +247,22 @@ class TestRunner {
         aggregatedResults.success =
           aggregatedResults.numFailedTests === 0 &&
           aggregatedResults.numRuntimeErrorTestSuites === 0;
-        if (reporter.onRunComplete) {
-          reporter.onRunComplete(config, aggregatedResults);
-        }
-        return aggregatedResults;
+
+        return this._hasteMap.matchFiles(/\.snap/).then(res => {
+          aggregatedResults.snapshotsUnlinked = 0;
+          res.forEach(snapFile => {
+            const parts = snapFile.split('__snapshots__/');
+            const testFile = parts[0] + parts[1].slice(0, -5);
+            if (!fileExists(testFile)) {
+              fs.unlinkSync(snapFile);
+              aggregatedResults.snapshotsUnlinked++;
```

Comment:
This happen whenever a test file gets deleted but not the associated `.snap` file.
I think this check can only happen after all the test have finished as there may be some cleanup stuff in afterAll that deletes files, like in integration tests, we probably don't want to mess around with those, so I'm not sure how to trigger a test failure here if not by adding a failure to the `aggregatedResults`, but that may be confusing as there will no be real test failure, that's why I think silently removing it would be ok